### PR TITLE
Setting the UserId on creation of connections and integrations #718

### PR DESCRIPTION
**Pull request by KurtStam**

### CODE REVIEW

File: model/src/main/java/io/syndesis/model/integration/Integration.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ default Kind getKind() {
 
     Optional<String> getIntegrationTemplateId();
 
-    Optional<String> getUserId();
+    String getUserId();
```
Comment (jimmidyson) [inline model/src/main/java/io/syndesis/model/integration/Integration.java]: Doesn't this mean the UI will have to send the userId every time with the request?

Comment (KurtStam) [reply model/src/main/java/io/syndesis/model/integration/Integration.java]: Yes think so, which is why I asked for @gashcrumb's review to make sure this is ok.

Comment (zregvart) [reply model/src/main/java/io/syndesis/model/integration/Integration.java]: Should we trust the username provided by the UI? Can we add a default method so its populated from Spring Security if not set already?

Comment (KurtStam) [reply model/src/main/java/io/syndesis/model/integration/Integration.java]: Yeah I like the idea of setting it on the backend. If we do that we should not even have the UI worry about it.

Comment (jimmidyson) [reply model/src/main/java/io/syndesis/model/integration/Integration.java]: We can add that as a default value, but would require muddying dependencies by adding spring secuity as a dependency to model.

**Comment by rhuss** [comment]
Could we please hold this off until TP2 is released ? Thanks !

**Comment by KurtStam** [comment]
@rhuss Sure we can wait. 

@jimmidyson I changed the PR so it leaves userId optional but it gets set when connections and integrations are created using the UserPrincipal.

**Comment by KurtStam** [comment]
retest this please

File: rest/src/main/java/io/syndesis/rest/v1/operations/Creator.java [hunk 2]
```diff
@@ -30,7 +32,7 @@
     @POST
     @Produces(MediaType.APPLICATION_JSON)
     @Consumes("application/json")
-    default T create(@NotNull @Valid T obj) {
+    default T create(@Context SecurityContext sec, @NotNull @Valid T obj) {
```
Comment (jimmidyson) [inline rest/src/main/java/io/syndesis/rest/v1/operations/Creator.java]: Is this needed here?

Comment (jimmidyson) [reply rest/src/main/java/io/syndesis/rest/v1/operations/Creator.java]: Oh scratch that, it's used in subclass. Hmm not exactly liking this, but can't see a way round it I guess.

**Comment by KurtStam** [comment]
retest this please

**Pull request merged**